response.go

```diff
@@ -40,21 +45,22 @@ func marshalResponse(msg []byte) (resp response, err error) {
 		resp.data = result["data"]
 	}
 	err = nil
-	resp.requestid = j["requestId"].(string)
+	resp.requestId = j["requestId"].(string)
 	return
 }
 
 // saveResponse makes the response available for retrieval by the requester. Mutexes are used for thread safety.
 func (c *Client) saveResponse(resp response) {
 	c.respMutex.Lock()
```

Comment:
At a quick glance it looks like calls to this lock can be removed now that the code is using a concurrent-safe map. Should help improve performance.

Reply:
I will run some tests to test the impact, mean while I push the other changes